specification/iotcentral/data-plane/Microsoft.IoTCentral/preview/2021-04-30-preview/iotcentral.json

```diff
@@ -541,6 +549,52 @@
         "data"
       ]
     },
+    "JobBatch": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Whether batching is done on a specified number of devices or a percentage of the total devices.",
+          "type": "string",
+          "enum": [
+            "number",
+            "percentage"
+          ]
+        },
+        "value": {
+          "description": "The number or percentage of devices on which batching is done.",
+          "type": "number"
+        }
+      },
+      "required": [
+        "type",
+        "value"
+      ]
+    },
+    "JobCancellationThreshold": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Whether the cancellation threshold is per a specified number of devices or a percentage of the total devices.",
+          "type": "string",
+          "enum": [
+            "number",
+            "percentage"
+          ]
+        },
+        "value": {
+          "description": "The number or percentage of devices on which the cancellation threshold is applied.",
+          "type": "number"
+        },
+        "batch": {
+          "description": "Whether the cancellation threshold applies per-batch or to the overall job.",
+          "type": "boolean"
```

Comment:
Instead of boolean, consider enum. If you ever want more than 2 options, it is easy to add an enum, but a boolean locks you in place.

Reply:
Make sense, as here only 2 options for us, use or not use batch operation, will not be changed.